README.md
# SIR

Simple Image Resizer is a tool created for batch parallel resizing, rotate
and converting images with metadata support.

You can add frame or text and apply color and gradient filters on your images.

You can also manipulate on histogram.

Finally you can do this all activities on many images at the same time using
your multiple core CPU!


[![Build Status](https://travis-ci.org/marek629/SIR.svg)](https://travis-ci.org/marek629/SIR)


## Licence

This is free software distributed under GNU GPL v2 license.
Read [LICENSE](LICENSE) file for details.



## Supported image formats


### Read/write supported image formats

* BMP  - Windows Bitmap
* JPEG - Joint Photographic Experts Group
* PNG  - Portable Network Graphics
* PPM  - Portable Pixmap
* TIFF - Tagged Image File Format (Qt Image Formats Plugin required)
* XBM  - X11 Bitmap
* XPM  - X11 Pixmap


### Read only supported image formats

* GIF  - Graphic Interchange Format (optional)
* MNG  - Multiple-image Network Graphics (Qt Image Formats Plugin required)
* PBM  - Portable Bitmap
* PGM  - Portable Graymap
* SVG  - Scalable Vector Graphics
* TGA  - Targa Image Format (Qt Image Formats Plugin and 3rd party TGA library required)


### Supported metadata standards using exiv2

* Exif (edition supported)
* IPTC (edition supported)
* XMP



## Installation

Installation instruction was stored in [INSTALL](INSTALL.md) file.



## Dependences

* [Qt 4.7+ or Qt 5.0+](http://qt-project.org/)


### Optional dependences

* [libexiv2](http://exiv2.org/)
* [dcraw](http://www.cybercom.net/~dcoffin/dcraw/)


### Build dependences

* [CMake 2.8.3+](http://www.cmake.org/)


### Documentation build dependences

SIR documentation was created using [Doxygen](http://www.stack.nl/~dimitri/doxygen/).


CHANGELOG.md
# v3.2.2

## Marek Jędryka <jedryka89@gmail.com>

### Fri Aug 04 2023

* Added support for exiv2 v0.28.0 thanks Rafael Sachetto


# v3.2.1

## Marek Jędryka <jedryka89@gmail.com>

### Sun May 30 2021

* Fixed build scripts for Arch Linux based distros


# v3.2

## Marek Jędryka <jedryka89@gmail.com>

### Sun Feb 04 2018

* Fixed loading transparent image with custom background colour
* Fixed GIF presence at output file formats list
* Fixed translations of standard dialog buttons
* Updated Spanish translation thanks José M. Ciordia
* Updated Polish translation


# v3.1

## Marek Jędryka <jedryka89@gmail.com>

### Thu Feb 11 2016

* Changed format of generated thumbnails from TIFF to JPEG
* Fixed wait currsor after unselect files issue
* Integrated tests with TravisCI platform
* Updated Polish translation (Raw basic tab)


# v3.0

## Marek Jędryka <jedryka89@gmail.com>

### Wed Jan 06 2016

* Added Raw tab into main window
* Added Basic view tab for user frendly Raw configuration
* Changed required C++ version to C++11


# v2.8

## Marek Jędryka <jedryka89@gmail.com>

### Thu Mar 05 2015

* Changed SIR to single instance application
* Changed minimum Qt 4.7 required
* Added status bar
* Added Dotate button in About dialog
* Improved saving "Exif.Image.Orientation" metadata tag
* Improved view in details for Aperture, Exposure Time and Exposure Bias metadata
* Improved keyboard navigation for selecting files in TreeWidget
* Removed setting empty dcraw path string after browse cancel
* Fixed left arrow shortcut for previous image in PreviewDialog
* Fixed no files with lowercase extension in "Add File" dialog
* Fixed enabling/disabling buttons in "File Details" options dialog after load settings
* Fixed refresh file details settings after save settings
* Fixed open link in default web browser  instead About dialog
* Fixed freeze of the application while loading many images
* Removed defines.h file
* Removed files coresponded uploading to Google Code
* CamelCase source files naming
* Updated Polish translation


# v2.7.3

## Marek Jędryka <jedryka89@gmail.com>

### Sun Aug 24 2014

* Changed all occurrences of old SIR webpage URL
* Windows: fixed button icons
* Disabled drag & drop URL checking for Qt 4.7 support


# v2.7.2

## Marek Jędryka <jedryka89@gmail.com>

### Sat May 31 2014

* Polish translation improved
* Spanish translation updated (thanks Garbage Collector)
* fixed language autosetting
* Windows: fixed Drag & Drop
* Windows: improved icon


# v2.7.1

## Marek Jędryka <jedryka89@gmail.com>

### Thu Apr 17 2014

* fixed translating of Gradient Effect widgets
* fixed initialization of Gradient Effect
* patched finding icon theme in Qt5 (Tango, oxygen and gnome themes are supported)


# v2.7

## Marek Jędryka <jedryka89@gmail.com>

### Fri Feb 28 2014

* added filter effect
* added disabling SVG tab if no SVG file loaded on file tree view and the view is not empty
* added Version class
* added effects storage in XML files
* added histogram stretching and equalization effects
* added clearing file list before session restoring
* added initializing session and effects directory based restored session from command line
* improved loading files contains incomplete/broken metadata
* CMake: build using choosed Qt version (4/5)
* code update for Qt 5 compatibility
* bugfix: painting effect when desired size typed in Bytes now
* String class moved from MetadataUtils namespace to sir namespace
* Polish translation updated


# v2.6

## Marek Jędryka <jedryka89@gmail.com>

### Sun Apr 28 2013

* added command line arguments --help, --session and --lang support
* added background color in convertion options
* added SVG convertion tab: remove single text node and group of text node containing specifed string
* added adding frame, text and picture effect and effect tab
* added sessions storage in XML files
* create empty metadata when converting from SVG file
* set current date time in Exif if needed
* iconsets moved from UI to C++ code for uic<4.8 compatibility
* Polish translation updated


# v2.5.1

## Marek Jędryka <jedryka89@gmail.com>

### Wed Nov 14 2012

* relative path bug-fix


# v2.5

## Marek Jędryka <jedryka89@gmail.com>

### Tue Nov 13 2012

* project moved to CMake
* Conditional compilation: disable or enable metadata support
* Settings and DetailsBrowser classes added
* Image width and height line edit replaced by spin box
* faster selection using Exiv2 image size detection if it's available
* bugfix: delete selected items from list in ConvertDialog - now delete all selected items
* shared data manipulation methods moved to SharedInformation;
* save and enabled fields in MetadataUtils::Metadata removed - now used Settings fields
* DetailsOptions namespace moved to separated file
* widgets moved to separated directory
* option and selection dialogs splited to few widgets for more flexibility
* remove temporary files at end of session
* width of SVG images render in DetailsBrowser patch
* drop event improved
* myQTreeWidget renamed to TreeWidget, some code moved from ConvertDialog
* external columns into tree view like size (px,K/MiB) added
* new options group box "File list" and header view context menu for TreeWidget settings
* converting from SVG bug when dest size is more than default size fixed
* keep aspect ratio added to settings
* English translation added, Polish translation updated
* many files moved from resource file to another files


# v2.4

## Marek Jędryka <jedryka89@gmail.com>

### Tue Jul 31 2012

* Changelog file provided
* Selected file (or files) details in splitter on rigth option added
* Easier translations of about dialog, about dialog corrected
* Droped QObject::tr() translated strings, non QObject inherited classes are using Q_DECLARE_TR_FUNCTIONS() macro now
* Added selection/import dialog
* Added Selection category in options dialog containing a few settings
* Bugfix loading preview from Exif metadata
* Added main page in documentation

blog.js
// Blog Data

var blogList = [
    {   'date': new Date(2023, 7, 4),
        'file': 'release_3.2.2.html',
        'author': 'marek'
    },
    {   'date': new Date(2021, 4, 30),
        'file': 'release_3.2.1.html',
        'author': 'marek'
    },
    {   'date': new Date(2018, 1, 4),
        'file': 'release_3.2.html',
        'author': 'marek'
    },
    {   'date': new Date(2016, 1, 11),
        'file': 'release_3.1.html',
        'author': 'marek'
    },
    {   'date': new Date(2016, 0, 6),
        'file': 'release_3.0.html',
        'author': 'marek'
    },
    {   'date': new Date(2015, 2, 5),
        'file': 'release_2.8.html',
        'author': 'marek'
    },
    {   'date': new Date(2015, 0, 6),
        'file': 'video_2.1.1_to_2.5.1.html',
        'author': 'marek'
    },
    {   'date': new Date(2014, 7, 24),
        'file': 'release_2.7.3.html',
        'author': 'marek'
    },
    {   'date': new Date(2014, 4, 31),
        'file': 'release_2.7.2.html',
        'author': 'marek'
    },
    {   'date': new Date(2014, 3, 24),
        'file': 'installer_2.7.1.html',
        'author': 'marek'
    },
    {   'date': new Date(2014, 3, 17),
        'file': 'release_2.7.1.html',
        'author': 'marek'
    },
    {   'date': new Date(2014, 3, 10),
        'file': 'new_site.html',
        'author': 'marek'
    },
    {   'date': new Date(2014, 1, 28),
        'file': 'release_2.7.html',
        'author': 'marek'
    },
    {   'date': new Date(2013, 3, 28),
        'file': 'release_2.6.html',
        'author': 'marek'
    },
    {   'date': new Date(2012, 10, 14),
        'file': 'release_2.5.1.html',
        'author': 'marek'
    },
    {   'date': new Date(2012, 10, 13),
        'file': 'release_2.5.html',
        'author': 'marek'
    }
]

var monthName = [
    'January', 'February', 'March',
    'April', 'May', 'June',
    'July', 'August', 'September',
    'October', 'November', 'December'
];

// Blog Logic

var getEntryPage = function(entry) {
    $.ajax({
        type: 'GET',
        async: false,
        url: entry.file,
        dataType: 'text',
        success: function (content) {
            $('#content').append(content);
            var date = entry.date;
            var timeTag = $('#content .entry-date').last();
            timeTag.text(
                monthName[date.getMonth()] + ' ' + date.getDate() + ', ' + date.getFullYear()
            );
            timeTag.attr(
                "datetime",
                date.toDateString()
            );
        }
    });
};

var showBlog = function() {
    $('#content').empty();

    var yearArray = new Array();
    var activeYear = 0;

    var hash = window.location.hash.replace('#', '');
    if (hash.length == 0)
        activeYear = blogList[0].date.getFullYear();
    else {
        var int = parseInt(hash);
        if (isNaN(int))
            activeYear = blogList[0].date.getFullYear();
        else
            activeYear = int;
    }

    $.each(blogList, function(i, entry) {
        var year = entry.date.getFullYear();

        if ($.inArray(year, yearArray) < 0) {
            yearArray.push(year);
            $('#menu-box').append(
                '<li>' +
                    '<a href="#' + year + '">' + year +
                        '<span class="badge pull-right">0</span>' +
                    '</a>' +
                '</li>'
            );
        }

        var yearBadge = $('#menu-box li:contains("' + year + '") span.badge');
        var count = parseInt(yearBadge.text());
//         count += 1;
        yearBadge.text(count + 1);

        if (year == activeYear)
            getEntryPage(entry);
    });

    $('#menu-box li:contains("' + activeYear + '")').first().addClass('active');
};

var compareBlogEntry = function(a, b) {
    return b.date.valueOf() - a.date.valueOf();
};

$().ready( function() {
    $.pathchange.init();

    blogList.sort(compareBlogEntry);

    $('#content').empty();
    showBlog();
    onWindowResize();
});

$(window).resize( onWindowResize );
function onWindowResize() {
    var menuBox = $('#menu-box');
    if ($(window).width() > 768) menuBox.addClass('nav-stacked');
    else menuBox.removeClass('nav-stacked');
}

$(window).bind('pathchange', function() {
    $('#content').empty();
    $('#menu-box').empty();
    showBlog();
});
